### PR TITLE
Move update_packages.dart to `flutter update-packages`

### LINKED ISSUE #1906: Can no longer run pub get or dev/update_packages.dart
Since we don't require that developers install Dart anymore, and we don't suggest that they put our copy of Dart in their path, they can no longer run `pub` or dart scripts like `dev/update_packages.dart`. This means they can no longer run the examples, for example.

cc @abarth

### PR DESCRIPTION
**Pull request by abarth**
Now that we don't require the Dart SDK to be in your path, it's hard to run
./dev/update_packages.dart. Instead, you can now run `flutter update-packages`.

Fixes #1906

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  @override
+  Future<int> runInProject() async {
+    Stopwatch timer = new Stopwatch()..start();
+    List<String> pubArgs = [ argResults['upgrade'] ? 'upgrade' : 'get' ];
+    int count = 0;
+    count += _runPub(new Directory("${ArtifactStore.flutterRoot}/packages"), pubArgs);
+    count += _runPub(new Directory("${ArtifactStore.flutterRoot}/examples"), pubArgs);
+    String command = "${sdkBinaryName('pub')} ${pubArgs.join(' ')}";
+    print("Ran \"$command\" $count times in ${timer.elapsedMilliseconds} ms");
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: `time${ count == 1 ? '' : 's' }`

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      'upgrade',
+      help: 'Run "pub upgrade" rather than "pub get".',
+      defaultsTo: false
+    );
+  }
+
+  @override
+  Future<int> runInProject() async {
+    Stopwatch timer = new Stopwatch()..start();
+    List<String> pubArgs = [ argResults['upgrade'] ? 'upgrade' : 'get' ];
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: `<String>[...]`

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+class UpdatePackagesCommand extends FlutterCommand {
+  final String name = 'update-packages';
+  final String description = 'Update the packages inside the Flutter repo.';
+
+  bool get hidden => _hideCommand;
+  final bool _hideCommand;
+
+  bool get requiresProjectRoot => false;
+
+  UpdatePackagesCommand({ hideCommand: false }) : _hideCommand = hideCommand {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: constructor first

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import '../artifacts.dart';
+import '../runner/flutter_command.dart';
+import '../base/process.dart';
+
+int _runPub(Directory directory, List<String> pubArgs) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: don't we have a "run pub" feature in the tools already?

**Comment by Hixie** [comment]
@devoncarew is more up to date on tools conventions these days (e.g. print vs stdout.write)

File: packages/flutter_tools/lib/src/commands/update_packages.dart (hunk 4 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: Yeah, `lib/src/dart/pub.dart`.

File: packages/flutter_tools/lib/src/commands/update_packages.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: `print` ==> `printStatus`

**Comment by devoncarew** [comment]
lgtm; we should probably move this to use the existing `pubGet()` method.

**Comment by abarth** [comment]
> we should probably move this to use the existing pubGet() method.

Done.

**Pull request merged**